doc/languages-frameworks/lisp.section.md

```diff
@@ -0,0 +1,225 @@
+# Lisp {#lisp}
+
+## Overview {#lisp-overview}
+
+The Lisp infrastructure works in the following way:
+
+1. Package definitions are imported from Quicklisp (`ql-import.lisp`)
+2. Missing native libraries etc. are fixed in `ql.nix`
+3. Additional packages are defined using `build-asdf-system` in `packages.nix`
+4. CL implementations are enhanced with the `pkgs` and `withPackages` attributes (`makeLisp`)
+5. Packages are overridden individually using `overrideLispAttrs`
+6. Packages are woven into a new scope using `packageOverlays`
+
+## Importing from Quicklisp {#lisp-importing-from-quicklisp}
+
+### The `ql-import` script {#lisp-ql-import}
+
+There is a Common Lisp program in the `import` directory, which downloads the
+`systems.txt` and `releases.txt` files from `quicklisp.org` and creates a Nix
+expression with package definitions in a file called `imported.nix`.
+
+To run this import process, execute:
+
+```sh
+$ nix-shell
+$ sbcl --script ql-import.lisp
+```
+
+It also creates a `packages.sqlite` file. It's used during the
+generation of the `imported.nix` file and can be safely removed. It
+contains the full information of Quicklisp packages, so you can use it
+to query the dependency graphs using SQL, if you're interested.
+
+### Choosing a Quicklisp release {#lisp-choosing-a-quicklisp-release}
+
+Quicklisp release URL is configured in the `import/main.lisp` file.
+
+Change the `:dist-url` keyword argument to `make-instance 'quicklisp-repository`
+before running `ql-import.lisp` to upgrade to a newer release.
```

Comment:
So the Quicklisp release is now neither in Nix code (with checksums) nor auto-bumpable (with `:latest` or whatever)?

Reply:
correct,
Do you think there should be checksumming  and auto update mechanism? It would be easy to add

Reply:
It feels more like «one way or another»: checksummed version choice available for Nix introspection looks natural, pick-version-trust-upstream looks also natural for a task like that, if «latest» is an option (where checksumming won't really give you much). Something in-between looks like having benefits of neither side…